java/server/src/org/openqa/selenium/grid/graphql/Node.java

```diff
@@ -26,14 +26,18 @@
 
   private final UUID id;
   private final URI uri;
+  private final String host;
   private final boolean isUp;
   private final int maxSession;
+  private final int port;
 
   public Node(UUID id, URI uri, boolean isUp , int maxSession) {
     this.id = Require.nonNull("Node id", id);
     this.uri = Require.nonNull("Node uri", uri);
     this.isUp = isUp;
     this.maxSession = Require.nonNull("Node maxSession", maxSession);
+    this.host = uri.getHost();
```

Comment:
Don't keep the values of derived fields as fields themselves unless calculating them is expensive. And we don't need this in this case.